utils/remote.cc

```diff
@@ -1455,7 +1458,31 @@ static void printTorrentList(tr_variant* top)
             "Status",
             "Name");
 
-        for (size_t i = 0, n = tr_variantListSize(list); i < n; ++i)
+        size_t num_torrents = tr_variantListSize(list);
+
+        std::vector<tr_variant*> tptrs;
+        tptrs.reserve(num_torrents);
```

Comment:
This does not seem to be a good fix: not just the space in memory without any valid elements is preallocated here, but actually a `num_torrents` number of empty (zeroed) elements are added -- as a result, sorting doesn't work anymore. We can then insert elements at the index instead of .push_back() them -- but finally what would be the gain?

Reply:
@lvd2 actually this brings up a question I'd missed, what is the `tr_variantDictFindInt(d, TR_KEY_id, nullptr)` check for? It doesn't seem to be in the previous code -- is there ever a time where this check can fail?

Reply:
> This does not seem to be a good fix: not just the space in memory without any valid elements is preallocated here, but actually a `num_torrents` number of empty (zeroed) elements are added -- as a result, sorting doesn't work anymore. We can then insert elements at the index instead of .push_back() them -- but finally what would be the gain?

Yes you are right, a screw fell of from my head when I wrote this.

Reply:
> @lvd2 actually this brings up a question I'd missed, what is the `tr_variantDictFindInt(d, TR_KEY_id, nullptr)` check for? The existing code doesn't seem to check for it and it's not needed for the sort function, so I'm curious what made this necessary?

I've never investigated it properly, besides replicating the simple already existing check for the presence of TR_KEY_id: https://github.com/transmission/transmission/blob/432fc9bafcd9765495fd8ca52cbcb2830700ccf3/utils/remote.cc#L1486 

Regarding the .reserve() for the std::vector, my general assumpltion is that while there are considerably less torrents (well, *usually*) than say a million, and since the transmission-remote is still a one-off console app (no need to run for hours/days/years as a daemon), there is very little sense in that .reserve() optimization, as well equally little sense in removing it if it is already there.